wordpress-mobile/WordPress-iOS
# Reader Comments: Show separator for last leaf comments

Refs #17476

This PR implements separator logic for the comment thread. Additionally, the separator is always hidden for the last row (as per the design).

I couldn't find a better term for this 😅 , but a "last leaf" comment is the last descendant of a top-level comment. I've also added a note in the code, but here's an example:

```
- comment 1
  - comment 2
    - comment 3 --> this is the last leaf of comment 1.
- comment 4
  - comment 5
    - comment 6
  - comment 7 --> this is the last leaf of comment 4.
- comment 8
```

## To Test

- Enable the `newCommentThread` feature flag.
- Go to Reader > any comment thread.
- 🔍  Verify that the cell separators are correctly shown for every "last leaf" comment.
- Rotate the device to landscape.
- 🔍 Verify that the cell separators are displayed correctly.

## Regression Notes
1. Potential unintended areas of impact
n/a. Feature is hidden behind a flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a. Feature is hidden behind a flag.

3. What automated tests I added (or what prevented me from doing so)
n/a. Feature is hidden behind a flag.

PR submission checklist:
